### PR TITLE
new numeric algorithms with new tests.

### PR DESCRIPTION
**Pull request by 1fisedi**
For factorial and binomial coefficient calculations.

### CODE REVIEW

File: Algorithms.Tests/Numeric/BinomialCoefficientTests.cs [hunk 1]
```diff
@@ -0,0 +1,35 @@
+﻿using System;
+using Algorithms.Numeric;
+using NUnit.Framework;
+
+namespace Algorithms.Tests.Numeric
+{
+    public static class BinomialCoefficientTests
+    {
+
```
Comment (siriak) [inline Algorithms.Tests/Numeric/BinomialCoefficientTests.cs]:
```suggestion
```

File: Algorithms/Numeric/BinomialCoefficient.cs [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+namespace Algorithms.Numeric
+{
+    /// <summary>
+    /// The binomial coefficients are the positive integers
+    /// that occur as coefficients in the binomial theorem.
+    /// </summary>
+    public static class BinomialCoefficient
+    {
+        /// <summary>
+        /// Calculates Binomial coefficients fort given input.
```
Comment (siriak) [inline Algorithms/Numeric/BinomialCoefficient.cs]:
```suggestion
        /// Calculates Binomial coefficients for given input.
```

File: Algorithms.Tests/Numeric/FactorialTests.cs [hunk 3]
```diff
@@ -0,0 +1,25 @@
+﻿using Algorithms.Numeric;
+using NUnit.Framework;
+
+namespace Algorithms.Tests.Numeric
+{
+    public static class FactorialTests
```
Comment (siriak) [inline Algorithms.Tests/Numeric/FactorialTests.cs]: Add tests for 0 (0! = 1) and negative numbers (exception).

File: Algorithms/Numeric/Factorial.cs [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+        /// <param name="num">Input number.</param>
+        /// <returns>Factorial of input number.</returns>
+        public static long Calculate(int num)
+        {
+            if (num < 0)
+            {
+                throw new ArgumentException("Only for num >= 0");
+            }
+
+            return num == 0 ? 1 : num == 1 ? 1 : num * Calculate(num - 1);
```
Comment (siriak) [inline Algorithms/Numeric/Factorial.cs]:
```suggestion
            return num == 0 ? 1 : num * Calculate(num - 1);
```
